### PR TITLE
Split up System.IO.FileSystem solution

### PR DESCRIPTION
**Pull request by Maxwe11**
As background for #746 `System.IO.FileSystem.sln` should be splitted into:
- System.IO.FileSystem.sln
- System.IO.FileSystem.Primitives.sln

### CODE REVIEW

File: src/System.IO.FileSystem/tests/packages.config [hunk 1]
```diff
@@ -1,7 +1,8 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="System.Diagnostics.Debug" version="4.0.10-beta-22512" targetFramework="portable-net45+win+wpa81" />
   <package id="System.IO" version="4.0.10-beta-22512" targetFramework="portable-net45+win+wpa81" />
+  <package id="System.IO.FileSystem.Primitives" version="4.0.0-beta-22605" targetFramework="portable-net45+win" />
```
Comment (ericstj) [inline src/System.IO.FileSystem/tests/packages.config]: We should be using a consistent build of nuget packages.

File: src/System.IO.FileSystem/src/packages.config [hunk 2]
```diff
@@ -6,6 +6,7 @@
   <package id="System.Diagnostics.Tools" version="4.0.0-beta-22512" targetFramework="portable-net45+win+wpa81" />
   <package id="System.Globalization" version="4.0.10-beta-22512" targetFramework="portable-net45+win+wpa81" />
   <package id="System.IO" version="4.0.10-beta-22512" targetFramework="portable-net45+win+wpa81" />
+  <package id="System.IO.FileSystem.Primitives" version="4.0.0-beta-22605" targetFramework="portable-net45+win" />
```
Comment (ericstj) [inline src/System.IO.FileSystem/src/packages.config]: Please use a consistent build of nuget packages for all references.

File: src/System.IO.FileSystem/tests/packages.config (hunk 1 above)
Comment (Maxwe11) [inline src/System.IO.FileSystem/tests/packages.config]: Could you clarify please. You mean that project reference should be used instead of nuget package?

Comment (ericstj) [inline src/System.IO.FileSystem/tests/packages.config]: No, Nuget package is fine, but please use the same build as the rest of the project: 22512.  Also can you make sure targetFramework is the same?  It's ok to copy/paste another package element (Nuget package manager isn't working well with our build time package resolution at the moment).

Comment (Maxwe11) [inline src/System.IO.FileSystem/tests/packages.config]: fixed

File: src/System.IO.FileSystem/src/packages.config (hunk 2 above)
Comment (Maxwe11) [inline src/System.IO.FileSystem/src/packages.config]: fixed

**Comment by weshaggard** [comment]
LGTM thanks for splitting these @Maxwe11

**Comment by ericstj** [comment]
LGTM

**Pull request merged**